designs/sam_local_inlinecode.rst

```diff
@@ -0,0 +1,82 @@
+.. contents:: **Table of Contents**
+   :depth: 2
+   :local:
+
+``InlineCode`` debug
+=====================
+This is the design for a feature to provide local testing capabilities to functions defined via **InlineCode** key schema.
+
+What is the problem?
+--------------------
+When new functions are started an absolute path of the folder/file containing AWS Lambda code package need to be passed
+to docker deamon in order to bind-mount the path inside the container running the AWs Lambda runtime.
+
+As of today you can only test locally your AWS Lambda Functions if defined via ``CodeUri`` key and not ``InlineCode``.
+
+What will be changed?
+---------------------
+In this proposal, we will change the code in a way that, if the tested AWS Lambda function is codified with
+``InlineCode`` key, we will create a temporary local file based on the ``InlineCode`` value, passing the temp
+file to bind-mount inside the container and make it possible to locally test ``InlineCode`` authoer Functions.
+
+Success criteria for the change
+-------------------------------
+#. Support all programming languages supported by ``InlineCode``:
```

Comment:
well, I'd say with the advent of custom runtimes this might open the door to other interpreted languages but, Java and friends should not be supported.. unless i missed something?

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-code.html#cfn-lambda-function-code-zipfile here only python and node are listed..